job-webhook/job_scraper.py

```diff
@@ -4,69 +4,99 @@
 Scrapes job postings from GitHub repository and sends updates via Discord webhook
 """
 
+import dataclasses
 import hashlib
 import json
 import logging
 import os
 import time
-from dataclasses import dataclass
 from datetime import datetime
 from typing import Optional
 
-import dataclasses
 import requests
 from bs4 import BeautifulSoup
 from dotenv import load_dotenv
 
 load_dotenv()
-MAGIC_LITERAL = "\u21b3"  # Magic literal for company name continuation
+COMPANY_NAME_CONTINUATION = "\u21b3"
 
 # Configure logging
 logging.basicConfig(
     level=logging.INFO,
-    format='%(asctime)s - %(levelname)s - %(message)s',
-    handlers=[
-        logging.FileHandler('job_scraper.log'),
-        logging.StreamHandler()
-    ]
+    format="%(asctime)s - %(levelname)s - %(message)s",
+    handlers=[logging.FileHandler("job_scraper.log"), logging.StreamHandler()],
 )
 logger = logging.getLogger(__name__)
 
-@dataclass
+
+@dataclasses.dataclass
 class JobPosting:
     """Represents a job posting with all relevant information."""
-    
+
     company: str
     role: str
     location: str
     application_link: str
     date_posted: str
-    
-    def to_dict(self) -> dict:
-        """Convert job posting to dictionary format."""
-        return {
-            'company': self.company,
-            'role': self.role,
-            'location': self.location,
-            'application_link': self.application_link,
-            'date_posted': self.date_posted
-        }
+
+
+@dataclasses.dataclass
+class JobCache:
+    """Represents cached job postings and content hash."""
+
+    content_hash: str
+    jobs: list[JobPosting]
+    last_updated: str
+
+
+@dataclasses.dataclass
+class EmbedField:
+    """Represents a field in a Discord embed."""
+
+    name: str
+    value: str
+    inline: bool = False
+
+
+@dataclasses.dataclass
+class EmbedFooter:
+    """Represents the footer of a Discord embed."""
+
+    text: str
+    icon_url: Optional[str] = None
+
+
+@dataclasses.dataclass
+class DiscordEmbed:
+    """Represents a Discord embed structure."""
```

Comment:
I would make it clear that this models a partial of [an actual Discord API type](https://discord.com/developers/docs/developer-tools/embedded-app-sdk#embed), so the reader knows such objects are ultimately destined as a request payload.

```suggestion
    """Represents an embed to post to the Discord API."""
```